tests/test_worker.py

```diff
@@ -124,8 +129,14 @@ def callback(channel, message):
         worker_group_t.start()
         # wait when a worker starts the callback and terminate the worker group
         callback_is_running.wait()
-        self.assertRaises(SystemExit, worker_group.sigterm_handler, None, None)
-        self.assertFalse(callback_is_running.is_set())
-
+        try:
+            worker_group.sigterm_handler(None, None)
+        except StopWorkerGroupLoop:
+            pass
```

Comment:
If this exception is always propagated, should we use `assertRaises` here?

Reply:
Well, the thing is that it's [not always propagated](https://travis-ci.org/django/channels/jobs/308991215). The problem here is that *usually* sub-worker threads start earlier and pick a message, so main worker is idle when we call sigterm handler and almost always there's an exception to break its loop; but [*sometimes*](https://github.com/django/channels/pull/792/files#diff-28a43fbdfeeb97555c5e29e15383a927R184) main worker picks it by itself, and calling sigterm handler doesn't bring the exception - we're humbly waiting when message is processed and the exception is not needed.

 This looks a bit tricky, so I improved the tests to cover all possible cases, please take a look.